src/services/WebViews.ts

```diff
@@ -409,26 +414,33 @@ export class WebViews {
 		
 		}	
 	}
-	private async resultStateChanged(selectedResultState: any, rows: any){
+	private async resultStateChanged(bulkComment: any, selectedResultState: any, rows: any){
 		let scanId = this.scanNode.scanId;
 		let nodes = this.queryNode.Result;
+
+		let mandatoryComment = CxSettings.getMandatoryCommentFlag();
+
+		const request = bulkComment === '' ? {"state" : selectedResultState} : {"state" : selectedResultState,"comment" : bulkComment};
 		//The below for loop updates the result state
 		for (var i = 0; i < rows.length; i++) {
 			var pathId = rows[i];
 			for (let nodeCtr = 0; nodeCtr < nodes.length; nodeCtr++) { 
 				if( pathId == nodes[nodeCtr].Path[0].$.PathId) {
-					let state = selectedResultState;
-					const request = {
-						"state" : state						
-					};
+					
 					try {
 						await this.httpClient.patchRequest(`sast/scans/${scanId}/results/${pathId}`, request);
-						nodes[nodeCtr].$.state = state;						
+						nodes[nodeCtr].$.state = selectedResultState;	
+						nodes[nodeCtr].$.Remark = bulkComment === '' ? nodes[nodeCtr].$.Remark  : `New Comment,${bulkComment}\r\n${nodes[nodeCtr].$.Remark}`;					
 					} 
 					catch (err) {
 						if (err.status == 404) {
 							this.log.error('This operation is not supported with CxSAST version in use.');
 						}
+						if(err.response.body.messageCode == 49797)
+						{
+							this.log.error("A mandatory comment is required while updating result state flag.");
```

Comment:
Remove the word 'mandatory' from the error message.

Reply:
Done